osu.Game/Updater/UpdateManager.cs

```diff
@@ -62,35 +66,44 @@ protected override void LoadComplete()
             // debug / local compilations will reset to a non-release string.
             // can be useful to check when an install has transitioned between release and otherwise (see OsuConfigManager's migrations).
             config.SetValue(OsuSetting.Version, version);
-        }
 
-        private readonly object updateTaskLock = new object();
+            config.BindWith(OsuSetting.ReleaseStream, releaseStream);
+            releaseStream.BindValueChanged(_ => CheckForUpdate());
 
-        private Task<bool>? updateCheckTask;
+            CheckForUpdate();
+        }
 
-        public async Task<bool> CheckForUpdateAsync()
+        /// <summary>
+        /// Immediately checks for any available update.
+        /// </summary>
+        public void CheckForUpdate()
         {
-            if (!CanCheckForUpdate)
-                return false;
-
-            Task<bool> waitTask;
-
-            lock (updateTaskLock)
-                waitTask = (updateCheckTask ??= PerformUpdateCheck());
+            _ = CheckForUpdateAsync();
+        }
 
-            bool hasUpdates = await waitTask.ConfigureAwait(false);
+        /// <summary>
+        /// Immediately checks for any available update.
+        /// </summary>
+        /// <returns><c>true</c> if any updates are available, <c>false</c> otherwise.</returns>
+        public async Task<bool> CheckForUpdateAsync(CancellationToken cancellationToken = default)
+        {
+            var cancellation = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+            var lastCancellation = Interlocked.Exchange(ref updateCancellation, cancellation);
 
-            lock (updateTaskLock)
-                updateCheckTask = null;
+            using (lastCancellation)
+            {
+                // This serves a dual purpose of nullifying the last update, closing any existing notifications as stale.
+                await lastCancellation.CancelAsync().ConfigureAwait(false);
+            }
```

Comment:
I battled internal demons here, and I'm still not sure of the best way to do this.

One alternative I considered is is asserting `ThreadSafety.IsUpdateThread` and doing away with the thread-safety. I quickly discarded that when I made the mistake of `Task.Run(CheckForUpdateAsync)` in the scheduled background check. 
That particular code instead uses the synchronous method above, but it's still something I consider easy to get wrong, so I settled on this somewhat convoluted way of handling thread safety. Hopefully it's not too hard to look at.

Reply:
I now consider the above comment moot, because I discovered the same being done by SSV2:

https://github.com/ppy/osu/blob/d03b5b7b0b196bce8f89435bff89017d7e974dc9/osu.Game/Graphics/Carousel/Carousel.cs#L319-L322

The difference here is `using (lastCancellation)`, which is required because this is a linked CTS.